### PR TITLE
fix serialization failure due to mismatched tags

### LINKED ISSUE #4588: AI fails with slow MP setup
Bug Report
==========

Environment
-----------

* **FreeOrion Version**: 63af30430242b3b1e02cf88f0fcec6666babf7ac
* **Operating System**: Linux
* **Fetched as** 
  * Compiled from source

Description
-----------

When launch test game with slow MP setup with AI, AIs fail to get game start message:

AI_1.log
```
23:27:20.506759 {0x00007f9b4221c780} [error] ai : Message.cpp:988 : ExtractGameStartMessageData(...) failed!  Message probably long, so not outputting to log.
Error: XML start/end tag mismatch - m_available_ship_parts
```

Expected Result
---------------

Slow MP game successfully starts with AI.

Steps to reproduce
------------------

* Put `auth.txt` file to config folder:

```
p
player_name:p:password
```

* Put `persistent_config.xml` to config folder:

```xml
<?xml version="1.0"?>
<XMLDoc>
 <save>
  <auto>
   <hostless>
    <enabled>1</enabled> <!-- Enable -->
    <each-player>
     <enabled>1</enabled> <!-- Enable -->
    </each-player>
   </hostless>
   <exit>
    <enabled>1</enabled>
   </exit>
   <interval>21600</interval>
  </auto>
 </save>
 <setup>
  <game>
    <uid>T001</uid>
  </game>
  <ai>
   <player>
    <count>1</count> <!-- Disable -->
   </player>
  </ai>
  <monster>
    <frequency>MONSTER_SETUP_LOW</frequency>
  </monster>
  <native>
    <frequency>GALAXY_SETUP_MEDIUM</frequency>
  </native>
  <specials>
    <frequency>GALAXY_SETUP_MEDIUM</frequency>
  </specials>
  <planet>
    <density>GALAXY_SETUP_MEDIUM</density>
  </planet>
  <galaxy>
   <shape>RING</shape>
   <age>GALAXY_SETUP_MEDIUM</age>
  </galaxy>
  <seed>RANDOM</seed>
  <star>
   <count>48</count>
  </star>
  <rules>
   <RULE_ENABLE_SUPER_TESTER>0</RULE_ENABLE_SUPER_TESTER>
   <RULE_ENABLE_EXPERIMENTORS>0</RULE_ENABLE_EXPERIMENTORS>
   <RULE_THRESHOLD_HUMAN_PLAYER_WIN>1</RULE_THRESHOLD_HUMAN_PLAYER_WIN>
   <RULE_ONLY_ALLIANCE_WIN>1</RULE_ONLY_ALLIANCE_WIN>
   <RULE_ALLOW_CONCEDE>1</RULE_ALLOW_CONCEDE>
   <RULE_CONCEDE_COLONIES_THRESHOLD>9999</RULE_CONCEDE_COLONIES_THRESHOLD>
   <RULE_SHIP_PART_BASED_UPKEEP>1</RULE_SHIP_PART_BASED_UPKEEP>
   <RULE_SHIP_HULL_COST_FACTOR>0.7</RULE_SHIP_HULL_COST_FACTOR>
   <RULE_SHIP_PART_COST_FACTOR>0.7</RULE_SHIP_PART_COST_FACTOR>
   <RULE_BUILDING_COST_FACTOR>0.7</RULE_BUILDING_COST_FACTOR>
   <RULE_TECH_COST_FACTOR>0.7</RULE_TECH_COST_FACTOR>
   <RULE_DIPLOMACY>RULE_DIPLOMACY_FORBIDDEN_FOR_ALL</RULE_DIPLOMACY>
   <RULE_ENABLE_ALLIED_REPAIR>1</RULE_ENABLE_ALLIED_REPAIR>
   <RULE_SINGULARITY_COST_FACTOR>0.5</RULE_SINGULARITY_COST_FACTOR>
   <RULE_NUM_COMBAT_ROUNDS>4</RULE_NUM_COMBAT_ROUNDS>
   <RULE_SHIP_STRUCTURE_FACTOR>8</RULE_SHIP_STRUCTURE_FACTOR>
   <RULE_SHIP_WEAPON_DAMAGE_FACTOR>6</RULE_SHIP_WEAPON_DAMAGE_FACTOR>
   <RULE_FIGHTER_DAMAGE_FACTOR>6</RULE_FIGHTER_DAMAGE_FACTOR>
   <RULE_EXTRASOLAR_SHIP_DETECTION>1</RULE_EXTRASOLAR_SHIP_DETECTION>
   <RULE_BASELINE_PLANET_STABILITY>2</RULE_BASELINE_PLANET_STABILITY>
   <RULE_PROTECTION_FOCUS_STABILITY>5</RULE_PROTECTION_FOCUS_STABILITY>
   <server-locked>
    <RULE_ENABLE_SUPER_TESTER>1</RULE_ENABLE_SUPER_TESTER>
    <RULE_ENABLE_EXPERIMENTORS>1</RULE_ENABLE_EXPERIMENTORS>
    <RULE_THRESHOLD_HUMAN_PLAYER_WIN>1</RULE_THRESHOLD_HUMAN_PLAYER_WIN>
    <RULE_ONLY_ALLIANCE_WIN>1</RULE_ONLY_ALLIANCE_WIN>
    <RULE_ALLOW_CONCEDE>1</RULE_ALLOW_CONCEDE>
    <RULE_CONCEDE_COLONIES_THRESHOLD>1</RULE_CONCEDE_COLONIES_THRESHOLD>
    <RULE_SHIP_PART_BASED_UPKEEP>1</RULE_SHIP_PART_BASED_UPKEEP>
    <RULE_SHIP_HULL_COST_FACTOR>1</RULE_SHIP_HULL_COST_FACTOR>
    <RULE_SHIP_PART_COST_FACTOR>1</RULE_SHIP_PART_COST_FACTOR>
    <RULE_BUILDING_COST_FACTOR>1</RULE_BUILDING_COST_FACTOR>
    <RULE_TECH_COST_FACTOR>1</RULE_TECH_COST_FACTOR>
    <RULE_DIPLOMACY>1</RULE_DIPLOMACY>
    <RULE_ENABLE_ALLIED_REPAIR>1</RULE_ENABLE_ALLIED_REPAIR>
    <RULE_SINGULARITY_COST_FACTOR>1</RULE_SINGULARITY_COST_FACTOR>
    <RULE_NUM_COMBAT_ROUNDS>1</RULE_NUM_COMBAT_ROUNDS>
    <RULE_SHIP_STRUCTURE_FACTOR>1</RULE_SHIP_STRUCTURE_FACTOR>
    <RULE_SHIP_WEAPON_DAMAGE_FACTOR>1</RULE_SHIP_WEAPON_DAMAGE_FACTOR>
    <RULE_FIGHTER_DAMAGE_FACTOR>1</RULE_FIGHTER_DAMAGE_FACTOR>
    <RULE_EXTRASOLAR_SHIP_DETECTION>1</RULE_EXTRASOLAR_SHIP_DETECTION>
    <RULE_BASELINE_PLANET_STABILITY>1</RULE_BASELINE_PLANET_STABILITY>
    <RULE_PROTECTION_FOCUS_STABILITY>1</RULE_PROTECTION_FOCUS_STABILITY>
   </server-locked>
  </rules>
 </setup>
 <network>
  <server>
   <ai>
    <max>1</max> <!-- Disable -->
   </ai>
   <allow-email>
    <new-turn>1</new-turn> <!-- Enable -->
   </allow-email>
   <allow-observers>1</allow-observers> <!-- Disable -->
   <send-turn-chat>1</send-turn-chat> <!-- Enable -->
   <human>
    <min>3</min>
   </human>
   <conn-human-empire-players>
    <min>0</min>
   </conn-human-empire-players>
   <unconn-human-empire-players>
    <max>0</max>
   </unconn-human-empire-players>
   <publish-statistics>1</publish-statistics> <!-- Disable in real game -->
   <publish-seed>1</publish-seed> <!-- Disable in real game -->
   <drop-empire-ready>0</drop-empire-ready>
   <cookies>
    <expire-minutes>5760</expire-minutes>
   </cookies>
   <binary>
    <enabled>0</enabled>
   </binary>
   <turn-timeout>
     <fixed-interval>0</fixed-interval>
     <max-interval>300</max-interval> <!-- 36h to hours = 129600 -->
   </turn-timeout>
   <client-message-size>
     <max>9485760</max>
   </client-message-size>
   <python>
     <asyncio-interval>1</asyncio-interval>
   </python>
  </server>
 </network>
 <logging>
  <sources>
   <network>debug</network>
  </sources>
 </logging>
</XMLDoc>
```

* Start server: `./freeoriond --hostless --quickstart`

### PR DESCRIPTION
**Pull request by geoffthemedio**
should fix  #4588

### CODE REVIEW

File: util/SerializeEmpire.cpp [hunk 1]
```diff
@@ -171,9 +171,9 @@ void Empire::serialize(Archive& ar, const unsigned int version)
             m_available_ship_hulls.insert(boost::container::ordered_unique_range, buf.begin(), buf.end());
 
         } else {
-            ar  & BOOST_SERIALIZATION_NVP(m_available_building_types)
-                & BOOST_SERIALIZATION_NVP(m_available_ship_parts)
-                & BOOST_SERIALIZATION_NVP(m_available_ship_hulls);
+            ar  & boost::serialization::make_nvp("m_available_building_types",m_available_building_types)
+                & boost::serialization::make_nvp("m_available_building_types", m_available_ship_parts)
```
Comment (o01eg) [inline util/SerializeEmpire.cpp]: Should `m_available_building_types` be repeated twice?

Comment (geoffthemedio) [reply util/SerializeEmpire.cpp]: it doesn't need to be, but it seemed odd to have just the first of three lines written differently

Comment (geoffthemedio) [reply util/SerializeEmpire.cpp]: oh, wait, you didn't mean what I thought...

Comment (geoffthemedio) [reply util/SerializeEmpire.cpp]: fixed, hopefully

**Review by o01eg — APPROVED** [review]
Test with slow MP setup succeeded.

**Pull request merged**